pyfolio/risk.py

```diff
@@ -0,0 +1,486 @@
+        - Example:
+                     Equity(24   Equity(62
+                       [AAPL])      [ABT])
+        2017-04-03	  -0.51284     1.39173
+        2017-04-04	  -0.73381     0.98149
+        2017-04-05	  -0.90132	   1.13981
+    '''
+
+    positions_wo_cash = positions.drop('cash', axis=1)
+    gross_exposure = positions_wo_cash.abs().sum(axis=1)
+
+    sfe = positions_wo_cash.multiply(risk_factor, level=1) \
+        .divide(gross_exposure, axis='index')
+    tot_sfe = sfe.sum(axis=1, skipna=True)
+
+    return tot_sfe
+
+
+def plot_style_factor_exposures(tot_sfe, factor_name, ax=None):
+    '''
+    Plots DataFrame output of compute_style_factor_exposures as a line graph
+
+    Parameters
+    ----------
+    tot_sfe : pd.Series
+        Daily style factor exposures (output of compute_style_factor_exposures)
+        - Time series with decimal style factor exposures
+        - Example:
+            2017-04-24    0.037820
+            2017-04-25    0.016413
+            2017-04-26   -0.021472
+            2017-04-27   -0.024859
+
+    factor_name : string
+        Name of style factor, for use in graph title
+    '''
+
+    if ax is None:
+        ax = plt.gca()
+
+    ax.plot(tot_sfe.index, tot_sfe, label=factor_name)
+    avg = tot_sfe.mean()
+    ax.axhline(avg, linestyle='-.', label='Mean = {:.3}'.format(avg))
+    ax.axhline(0, color='k', linestyle='-')
+    _, _, y1, y2 = plt.axis()
+    a = max(abs(y1), abs(y2))
+    ax.set_ylim(-a, a)
+    ax.set_title('Exposure to {}'.format(factor_name))
+    ax.set_ylabel('{} \n Weighted Exposure'.format(factor_name))
+    ax.legend()
+
+    return ax
+
+
+def compute_sector_exposures(positions, sectors, sector_dict=None):
+    '''
+    Returns arrays of long, short and gross sector exposures of an algorithm's
+    positions
+
+    Parameters
+    ----------
+    positions : pd.DataFrame
+        Daily equity positions of algorithm, in dollars.
+        - See full explanation in compute_style_factor_exposures.
+
+    sectors : pd.DataFrame
+        Daily Morningstar sector code per asset
+        - See full explanation in create_risk_tear_sheet
+
+    sector_dict : dict or OrderedDict
+        Dictionary of all sectors
+        - Keys are sector codes (e.g. ints or strings) and values are sector
+          names (which must be strings)
+        - Defaults to Morningstar sectors
+    '''
+
+    if sector_dict is None:
+        sector_ids = SECTORS.keys()
+    else:
+        sector_ids = sector_dict.keys()
+
+    long_exposures = []
+    short_exposures = []
+    gross_exposures = []
+    net_exposures = []
+
+    positions_wo_cash = positions.drop('cash', axis=1)
+    long_exposure = positions_wo_cash[positions_wo_cash > 0].sum(axis=1)
+    short_exposure = positions_wo_cash[positions_wo_cash < 0].abs().sum(axis=1)
+    gross_exposure = positions_wo_cash.abs().sum(axis=1)
+    sectors.columns = sectors.columns.astype(int)
+
+    for sector_id in sector_ids:
```

Comment:
I'm all for using long data, but I want to point out that if our input data structures change, this isn't the only piece of code that will need refactoring: all `compute_foo()` functions will have maintenance issues.